### ISSUE TITLE
Default customer and default project

### ISSUE THREAD

**Issue by geniusupgrader** [kevinpapst/kimai2#403]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when, I have to start a simple activity, which does not belong to any project, because the project is a mandatory field, and I need to enter something.

**Describe the solution you'd like**
I would like to see, a default customer and default project, which is set in the settings. So that you don't have to enter always the same customer or project. This is beneficial if your workflow only consists of one customer (maybe yourself) and/or if your workflow does not use projects at all (for example: default project: personal)

**Additional context**
Coming from #401

**Comment by kevinpapst** [comment]
Extending on that: we could make customer and project "optional" as long as there is only one of both existing. The **create timesheet** and **edit timesheet** form would fetch that one automatically and hide the select-boxes.

Everyone who wants to track only personal times could then simply create "myself" as customer and "private" as project. In the future an installer could even have the choice to pre-create this structure.

**Comment by geniusupgrader** [comment]
I have tested your PR #443 and it works as expected, many thanks. 
It is now a breeze to enter activities, when there are only one customer and one project.
But unfortunately it does not fix my #403 issue complety, as described in your pull request description. Because my issue also targets cases, where there are multiple customers and projects. Would there be an option to set a default customer and a default project (as proposed in my request), it will also work with more than one customer and one project.
I know you are reluctant to add extra options and want to simplify UX, please consider my thoughts and keep it up!

**Comment by kevinpapst** [comment]
Yep I know its not exactly what you asked for. But in this case I would like to question if we can't find a better approach.

- You can restart your recent activities from the toolbar directly, your favorite one is likely one of them. If it doesn't show up there,  you don't use it that often, so having it pre-selected might make it even worse.
- You can restart an activity right away from your timesheet view, your favorite action should be on the first page

The only "problem" I can see here: you don't get the chance to change start time or description in this case without the additional "edit record" click.

I am just questioning if its the best approach we can come up or if something like: 
- bookmarks your favorite activities
- your most used activities & projects
in an additional dropdown might be more helpful. 

Or we show a tiny modal to edit description and start time before the restart action is executed ...

**Comment by geniusupgrader** [comment]
Sorry for my late response.
I think we are talking about different things here.
In your last comment you're talking about activities, but I mean customer and projects.

Now, after your PR #443, when there are only one customer, the customer field is hidden. The project field is always prefilled with the first project.

**I want to choose a default project, which gets prefilled with (and not the first one..)**

**And when there are multiple customers, you need to choose a customer again (its empty at first). What I want is, that you can choose a default customer.**

(The bold sentences are, what I really wanted in the first place with this issue)

This has nothing to do with activities (because as you said: there is the toolbar for the last 10 recent activities)

**Comment by geniusupgrader** [comment]
My personal use case at the moment (still experimenting what is best with a personal setup):

- one customer: me
- two projects: me, meetings
- 15 (global) activities

With this setup, I'm totally satisfied, because the first project (me) is preselected (when creating a new time). This is because I've created the project „me“ before the project „meetings“. But, hypothetical, if the project „meeting“ would be preselected instead... There is no option to change that. I'm not just thinking about me here (because its ok for me now) but I think of the other kimai users and for the future. There should be an option to set a default customer and default projects if there are multiple customers and multiple projects. Example:

- three customers: me, you, him
- five projects: one, two, three, four, five
- 15 (global) activities

Now, we suppose, I mainly work for the customer „him“ in the project „two“ at the moment. So, I click on the big new time button, in the upper right corner, to create a new time. But wait...!, the customer field is  prefilled with the customer „me“ and the project field is prefilled with the project „one“. What a burden, everytime I create a new time (with this way, not the recent activities) I need to change the customer and the project to what I work with at the moment (which is customer „him“ and project „two“).

I think this is a valid use case, don't you think @kevinpapst ?

**Comment by kevinpapst** [comment]
I am not saying your request is not valid: I am questioning if this is the best solution we can find ;-)
You kind of said it already: using the recent activities menu instead would solve your "problem". I am not saying you have to adapt to Kimai, the software should support you in tracking your times. 
Again: I am really interested in the best solution. Lets see if someone else has an opinion as well.

**Comment by dzmudist** [comment]
In the past I used to work with home-grown time tracking system, where was ability to predefine default Customer/Project, such that when you create new report then Customer and Project fields got populated. It was convenient for those developers and qa who continue work on the same project for long time, because they mostly switch only between Activities like: development, code review, testing, investigation, etc, but for example they may interrupt for several hours or days to do work for other projects and customers, thus fields Customer and Project are still should be available for them to set other Customer and Project than specified in default value.
Meanwhile for those persons who deals with multiple projects in the same day (e.g architects and management) it was not enough to have just one default combination and these persons has to dig through tons of customers/projects and activities in the drop downs in order to find the one they need. But even so, within a period of time, like week or month, in 80% of cases they switch between small subset of Customers and Projects. In average I would say that 80% of time you switch between about 5-7 mutually exclusive combinations of Customer/Project/Activity.

**Feature Proposal:**
Having that in mind the feature that may be useful and covers "default value for Customer/Project/Activity" will be something like "Saved Favorite Combinations" where one of saved favorite combination can become "Default Favorite Combination". Here is user stories that attempts to explain what I meant: 
1) As a user I would like to save Customer+Project+Activity combinations as "favorite" such that when creating new record in timesheet I can select it from the list of saved favorite combinations as result selected "favorite combination" will fill Customer Project and Activity fields    
2) As a user I would like to make saved favorite combination as default such that when create new record this particular favorite combination populates Customer/Project/Activity fields.
3) As a user I would like to set custom label for saved favorite combination in order to make it easier to find required combination in the list of other saved favorite combinations because custom names would be shorter and better recognizable.

**Comment by bernd-wechner** [comment]
I'll just add that from a UX perspective, the web interface I find intolerably difficult to use as every time I start an activity I'm prompted to fill out a form selecting a project and activity. I'm now use Kemai the desktop app because it remembers so I can click stop, start etc at will and it just uses the same project and activity.

The main consideration is, that someone logging a time wants minimal interactions and focus shifts to achieve a desired outcome and there are only four use cases I experience:

1. Start a new activity (new project, and/or activity)
2. Restart a regular activity after a period of inactivity (something I'm working on a present, after a lunch break say)
3. Restart a regular activity after working on another activity (i.e I may be working on two or three interleaving jobs and want to swap from one to the other)
4. Stop an activity that's running

In each case I want to minimise the number of UI interactions (keystrokes, mouse moves, clicks) and focus shifts (moving my eyes around following stuff).

**Comment by kevinpapst** [comment]
I do appreciate positive feedback, but writing stuff like this makes me ignore/block you in the future.

> the web interface I find **intolerably** difficult to use

1. Obviously this feature request could not help.

2. and 3. require exactly two clicks:
- 1 for open "last activities dropdown", 1 to restart it 
- 1 for action dropdwn and 1 for copy/restart 
- or just even only 1 if using drag&drop in the calendar

4. Either 2 clicks if you stop manually or none if you start another record.

This is open source. Contribute if you think you can do it better.

**Comment by bernd-wechner** [comment]
Well, ignore and block away if a little feedback is something that triggers your nerves so badly. Not how I collect UX experience notes form the public (by telling them if they struggle not only is their experience invalid but they are not worth hearing), but no skin of my little ol' nose if that's the way you wanna roll. 

It's great to see a project like this rolling, it is.  Keep up the good work. And thanks for provide UX improvement tips. Will explore (though am mostly using Kemai right now).

**Comment by kevinpapst** [comment]
You don't seem to recognize how you communicate in MY HOME, after using my software that I sent hundreds of hours building.

No sorry, no feedback to my notes, that was to be expected. Bye!

**Comment by hn-matti** [comment]
is there any chance to get prefilled activity based on the choice of the Project?
Our workflow at the moment is with the shortlink `$URL/timesheet/create?project=322&activity=24` which is very static.

**Comment by sgsgmbh** [comment]
Same here, is a chance to get prefilled activity or favorite activity?

**Comment by muety** [comment]
I'd appreciate to have default customer, project and activity!

**Comment by Freundschaft** [comment]
same, im confused cause it seems that there was a feature like this in 2018, but in the current version it seems like I always have to select project and activity, even if there is just one defined?

**Comment by tknupf** [comment]
I agree I want this feature. Please add this.
